modules/snooker/src/main/ts/ephox/snooker/api/TableOperations.ts

```diff
@@ -161,28 +161,39 @@ const opMakeRowHeader = (grid: Structs.RowCells[], detail: Structs.DetailExt, co
 };
 
 const opMakeRowsHeader = (initialGrid: Structs.RowCells[], details: Structs.DetailExt[], comparator: CompElm, genWrappers: GeneratorsTransform) => {
-  const rows = uniqueRows(details);
-
-  const replacer = (currentGrid: Structs.RowCells[], row: Structs.DetailExt) => TransformOperations.replaceRow(currentGrid, row.row, comparator, genWrappers.replaceOrInit);
+  const replacer = (currentGrid: Structs.RowCells[], row: Structs.DetailExt) =>
+    TransformOperations.replaceRow(currentGrid, row.row, comparator, genWrappers.replaceOrInit);
 
+  const rows = uniqueRows(details);
   const newGrid = Arr.foldl(rows, replacer, initialGrid);
-
   return bundle(newGrid, details[0].row, details[0].column);
 };
 
 const opMakeColumnHeader = (initialGrid: Structs.RowCells[], detail: Structs.DetailExt, comparator: CompElm, genWrappers: GeneratorsTransform) => {
   const newGrid = TransformOperations.replaceColumn(initialGrid, detail.column, comparator, genWrappers.replaceOrInit);
-
   return bundle(newGrid, detail.row, detail.column);
 };
 
 const opMakeColumnsHeader = (initialGrid: Structs.RowCells[], details: Structs.DetailExt[], comparator: CompElm, genWrappers: GeneratorsTransform) => {
+  const replacer = (currentGrid: Structs.RowCells[], column: Structs.DetailExt) =>
+    TransformOperations.replaceColumn(currentGrid, column.column, comparator, genWrappers.replaceOrInit);
+
   const columns = ColUtils.uniqueColumns(details);
+  const newGrid = Arr.foldl(columns, replacer, initialGrid);
+  return bundle(newGrid, details[0].row, details[0].column);
+};
 
-  const replacer = (currentGrid: Structs.RowCells[], column: Structs.DetailExt) => TransformOperations.replaceColumn(currentGrid, column.column, comparator, genWrappers.replaceOrInit);
+const opMakeCellHeader = (initialGrid: Structs.RowCells[], detail: Structs.DetailExt, comparator: CompElm, genWrappers: GeneratorsTransform) => {
+  const newGrid = TransformOperations.replaceCell(initialGrid, detail.row, detail.column, comparator, genWrappers.replaceOrInit);
+  return bundle(newGrid, detail.row, detail.column);
+};
 
-  const newGrid = Arr.foldl(columns, replacer, initialGrid);
+const opMakeCellsHeader = (initialGrid: Structs.RowCells[], details: Structs.DetailExt[], comparator: CompElm, genWrappers: GeneratorsTransform) => {
+  const replacer = (currentGrid: Structs.RowCells[], detail: Structs.DetailExt) =>
+    TransformOperations.replaceCell(currentGrid, detail.row, detail.column, comparator, genWrappers.replaceOrInit);
 
+  const columns = ColUtils.uniqueColumns(details);
```

Comment:
I'd rename to `uniqueColumns` and `uniqueRows` below 😬

Reply:
Not sure what you mean here? Below it's called `rows` and `columns` everywhere. There is a `uniqueRows` function similar to the `uniqueColumns` function used here that you may have got confused by?

Reply:
Nope, I meant to change the name of the variable from `const columns` to `const uniqueColumns` but if you have a function with the same name just leave as it is 👍